packages/thirdweb/src/react/web/ui/prebuilt/thirdweb/ClaimButton/index.tsx

```diff
@@ -0,0 +1,195 @@
+    }
+  | {
+      type: "ERC1155";
+      quantity: bigint;
+      tokenId: bigint;
+    }
+  | ({
+      type: "ERC20";
+    } & ({ quantityInWei: bigint } | { quantity: string }));
+
+export type ClaimButtonProps = Omit<TransactionButtonProps, "transaction"> & {
+  contractAddress: Hex;
+  chain: Chain;
+  client: ThirdwebClient;
+  claimParams: ClaimParams;
+};
+
+/**
+ * This button is used to claim tokens (NFT or ERC20) from a given thirdweb Drop contract
+ * there are 3 type of Drop contract: NFT Drop (DropERC721), Edition Drop (DropERC1155) and Token Drop (DropERC20)
+ * Note: This button only works with thirdweb Drop contracts. For custom contract, please use TransactionButton
+ * @param props
+ * @returns A wrapper for TransactionButton
+ *
+ * @example
+ *
+ * ```tsx
+ * import { ClaimButton } from "thirdweb/react";
+ * import { ethereum } from "thirdweb/chains";
+ *
+ * // For NFT Drop (ERC721)
+ * <ClaimButton
+ *   contractAddress="0x..." // contract address of the NFT Drop
+ *   chain={ethereum}
+ *   client={client}
+ *   claimParams={{
+ *     type: "ERC721",
+ *     quantity: 1n, // claim 1 token
+ *   }}
+ * >
+ *   Claim now
+ * </ClaimButton>
+ *
+ *
+ * // For Edition Drop (ERC1155)
+ * <ClaimButton
+ *   contractAddress="0x..." // contract address of the Edition Drop
+ *   chain={ethereum}
+ *   client={client}
+ *   claimParams={{
+ *     type: "ERC1155",
+ *     quantity: 1n,
+ *     tokenId: 0n,
+ *   }}
+ * >
+ *   Claim now
+ * </ClaimButton>
+ *
+ *
+ * // For Token Drop (ERC20)
+ * <ClaimButton
+ *   contractAddress="0x..." // contract address of the Token Drop
+ *   chain={ethereum}
+ *   client={client}
+ *   claimParams={{
+ *     type: "ERC20",
+ *     quantity: "100", // claim 100 ERC20 tokens
+ *   }}
+ * >
+ *   Claim now
+ * </ClaimButton>
+ *
+ * ```
+ */
+export function ClaimButton(props: ClaimButtonProps) {
+  const { children, contractAddress, client, chain, claimParams } = props;
+  const contract = getContract({
+    address: contractAddress,
+    client,
+    chain,
+  });
+  const account = useActiveAccount();
+  return (
+    <TransactionButton
+      transaction={async () => {
+        switch (claimParams.type) {
+          case "ERC721": {
+            const [{ isERC721 }, { isClaimSupported }, { claimTo }] =
+              await Promise.all([
+                import("../../../../../../extensions/erc721/read/isERC721.js"),
+                import(
+                  "../../../../../../extensions/erc721/__generated__/IDrop/write/claim.js"
+                ),
+                import(
+                  "../../../../../../extensions/erc721/drops/write/claimTo.js"
+                ),
+              ]);
+            const [is721, claimSupported] = await Promise.all([
+              isERC721({ contract }),
+              isClaimSupported(contract),
+            ]);
+            if (!is721 || !claimSupported) {
+              throw new Error("Not a valid NFT Drop (ERC721) contract");
+            }
```

Comment:
these should be different errors so that the user does not get confused - one is "this is not an erc721" - one is "this IS an erc721 but it doesn't support "claim" functionality

we can word that better and potentially even link to the docs for the second one (have the link in the error message potentially)?